content/200-orm/100-prisma-schema/20-data-model/90-multiple-files.mdx

```diff
@@ -0,0 +1,71 @@
+---
+
+:::tip
+
+Multi-file Prisma schema is available via the `prismaSchemaFolder` preview feature in Prisma versions 5.15.0 and later.
+
+:::
+
+## How to enable multi-file Prisma schema support
+
+Multiple Prisma schema file support is currently in preview. To enable the feature, add the `prismaSchemaFolder` feature flag to the `previewFeatures` field of the `generator` block in your Prisma schema: 
+
+```prisma file=schema.prisma showLineNumbers
+generator client {
+  provider        = "prisma-client-js"
+  //add-next-line
+  previewFeatures = ["prismaSchemaFolder"]
+}
+
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+```
+
+## How to include multiple Prisma schema files in your project
+
+To use multiple Prisma schema files, add a `schema` inside of your current `prisma` directory. With the `prismaSchemaFolder` Preview feature enabled, you can add as many files as you want to the `prisma/schema` directory.
+
+## How to use existing Prisma CLI commands with multiple Prisma schema files
+
+Existing Prisma CLI commands have been updated to support multiple Prisma files. Simply replace references to a file with a directory. As an example, the following `prisma migrate diff` command:
+
+```terminal
+npx prisma migrate diff \
+--from-empty \
+--to-schema-datamodel prisma/schema.prisma \
+--script > prisma/migrations/0_init/migration.sql
+```
+
+becomes the following:
+
+```terminal
+npx prisma migrate diff \
+--from-empty \
+//highlight-next-line
+--to-schema-datamodel prisma/schema \
+--script > prisma/migrations/0_init/migration.sql
+```
+
+## Best practices
+
+We’ve found that a few best practices work well with this new feature and will help you get the most out of it:
+
+1. Organize your files by domain: group related models into the same file. For example, keep all user-related models in `user.prisma` while post-related models go in `post.prisma`. Try to avoid having “kitchen sink” schema files.
+
+1. Use clear naming conventions: schema files should be named clearly and succinctly. Use names like `user.prisma` and `post.prisma` and not `myModels.prisma` or `CommentFeaturesSchema.prisma`.
+
+1. Have an obvious “main” schema file: while you can now have as many schema files as you want, you’ll still need a place where you define `datasource` and `generator` blocks. We recommend having a single schema file that’s obviously the “main” file so that these blocks are easy to find. `main.prisma`, `schema.prisma`, and `base.prisma` are a few we’ve seen that work well.
+
+## Examples
+
+Our fork of [dub by dub.co](https://github.com/prisma/dub) is a great example of a real world project adapted to use multiple Prisma schema files.
+
+## Learn more about the `prismaSchemaFolder` preview feature
+
+To learn more about future plans for the `prismaSchemaFolder` preview feature, or to give feedback, refer to [our Github issue](https://github.com/prisma/prisma/issues/1122).
```

Comment:
No future plans in that issue, and for feedback we will have the preview feature discussion tomorrow shortly before release.

Reply:
the issue is filler until we have that discussion.
```suggestion
To give feedback on the `prismaSchemaFolder` Preview feature, please refer to [our dedicated Github discussion](https://github.com/prisma/prisma/issues/1122).
```